protostar/cheatable_starknet/cheatcodes/expect_call_cairo_cheatcode.py

```diff
@@ -0,0 +1,48 @@
+import asyncio
+from typing import Callable, Optional
+
+from protostar.starknet import CheatcodeException
+from protostar.starknet import RawAddress, Address
+from protostar.starknet.cheater import CheaterException
+from protostar.starknet.data_transformer import CairoOrPythonData
+from protostar.cheatable_starknet.cheatcodes.cairo_cheatcode import CairoCheatcode
+
+
+class ExpectCallCairoCheatcode(CairoCheatcode):
+    @property
+    def name(self) -> str:
+        return "expect_call"
+
+    def build(
+        self,
+    ) -> Callable[[RawAddress, str, Optional[CairoOrPythonData]], Callable]:
+        return self.expect_call
+
+    def expect_call(
+        self,
+        contract_address: RawAddress,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> Callable:
+        return asyncio.run(
+            self._expect_call(
+                contract_address=Address.from_user_input(contract_address),
+                function_name=function_name,
+                calldata=calldata,
+            )
+        )
+
+    async def _expect_call(
+        self,
+        contract_address: Address,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> Callable:
+        try:
+            return await self.cheaters.expects.expect_call(
+                contract_address=contract_address,
+                function_name=function_name,
+                calldata=calldata,
+            )
+        except CheaterException as exc:
```

Comment:
Can we not catch CheaterException here, but specific errors raised from cheaters? It is definition of user-interface and the decision should be defined in Cheatcode logic

Reply:
Sorry, but I don't understand what's wrong with catching the parent class error here. From some places, we're going to receive just `CheaterException`, for example from `transform_calldata_to_cairo_data`.

> the decision should be defined in Cheatcode logic

what decision? I understand that we just get an error here and we rethrow so the error message is accessible higher in the stack.

If you'd prefer to make changes to the error-handling mechanism maybe that should be placed in a separate PR.

Reply:
It can be done in separate PR if you prefer to do so, but it would be simpler to include it here. I provided more extensive argument behind this/explanation on slack, I probably should let you know about it here, sorry 😞

Reply:
Okay, I think we should wait until #1404 is merged